Adding an aggregation to AllMatches search causes panic

When you add an aggregation to `AllMatches` it panics due to assignment to nil map.

test case 

```golang
func TestAllMatchesWithAggregation(t *testing.T) {
	query := NewMatchQuery("bluge").SetField("name")
	request := NewAllMatches(query)

	request.AddAggregation("score", aggregations.MaxStartingAt(search.DocumentScore(), 0))
}
```

result: 
```
╰─➤  go test -run AllMatches                                                                  1 ↵
--- FAIL: TestAllMatchesWithAggregation (0.00s)
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map

goroutine 6 [running]:
testing.tRunner.func1.1(0x1265fc0, 0x12f63b0)
	/usr/local/Cellar/go/1.15.2/libexec/src/testing/testing.go:1076 +0x30d
testing.tRunner.func1(0xc000001380)
	/usr/local/Cellar/go/1.15.2/libexec/src/testing/testing.go:1079 +0x41a
panic(0x1265fc0, 0x12f63b0)
	/usr/local/Cellar/go/1.15.2/libexec/src/runtime/panic.go:969 +0x175
github.com/blugelabs/bluge/search.Aggregations.Add(...)
	/Users/voldyman/dev/bluge/search/aggregations.go:29
github.com/blugelabs/bluge.(*AllMatches).AddAggregation(...)
	/Users/voldyman/dev/bluge/search.go:200
github.com/blugelabs/bluge.TestAllMatchesWithAggregation(0xc000001380)
	/Users/voldyman/dev/bluge/search_test.go:1270 +0x105
testing.tRunner(0xc000001380, 0x12bb3c0)
	/usr/local/Cellar/go/1.15.2/libexec/src/testing/testing.go:1127 +0xef
created by testing.(*T).Run
	/usr/local/Cellar/go/1.15.2/libexec/src/testing/testing.go:1178 +0x386
exit status 2
FAIL	github.com/blugelabs/bluge	0.080s
```

--- a/AUTHORS
+++ b/AUTHORS
@@ -8,3 +8,4 @@
 # Please keep the list sorted.
 
 Marty Schoch <marty.schoch@gmail.com>
+Akshay Shekher <akshay.shekher@gmail.com>

--- a/search.go
+++ b/search.go
@@ -184,7 +184,8 @@ type AllMatches struct {
 func NewAllMatches(q Query) *AllMatches {
 	return &AllMatches{
 		BaseSearch: BaseSearch{
-			query: q,
+			query:        q,
+			aggregations: make(search.Aggregations),
 		},
 	}
 }

--- a/search_test.go
+++ b/search_test.go
@@ -21,6 +21,7 @@ import (
 	"strconv"
 	"testing"
 
+	"github.com/blugelabs/bluge/search/aggregations"
 	"github.com/blugelabs/bluge/search/highlight"
 
 	"github.com/blugelabs/bluge/analysis/char"
@@ -1260,4 +1261,13 @@ func TestSearchHighlightingWithRegexpReplacement(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+}
+
+func TestAllMatchesWithAggregationIssue31(t *testing.T) {
+	query := NewMatchQuery("bluge").SetField("name")
+	request := NewAllMatches(query)
+
+	// This line would panic because aggregations map was not initialized internally
+	// should not panic with the fix
+	request.AddAggregation("score", aggregations.MaxStartingAt(search.DocumentScore(), 0))
 }

